src/components/access-settings/AccessSettings.tsx

```diff
@@ -256,18 +273,28 @@ const AccessSettings: FunctionComponent<{study: Study}> = ({study}) => {
   }
 
   const inviteUser = async (account: NewOrgAccount) => {
+    setIsCreatingNewMember(true)
     if (!account.email) {
       setNewOrgAccount({...account, error: 'No email provided'})
+      setIsCreatingNewMember(false)
       return
     }
-    const [success, error] = await createNewAccount(account.email, account.access, token!, orgMembership!)
+    const [success, error] = await createNewAccount(
+      account.email,
+      account.firstName,
+      account.lastName,
+      account.access,
+      token!,
+      orgMembership!
+    )
```

Comment:
I feel like it's a weird pattern to swallow the error in `createNewAccount` but still return it here. I was expecting you would need `try ... finally { setIsCreatingNewMember(false) }` but I see that `createNewAccount` can never throw, so this is logically fine. Just a little weird.

If you have react-query in this project, I would suggest capturing `createNewAccount` in a custom hook that wraps `useMutation`, then that hook will just provide state to replace `isCreatingNewMember` and the error.

Reply:
Good idea! Done in the next commit